keps/sig-multicluster/4322-cluster-inventory/README.md

```diff
@@ -506,6 +509,30 @@ With recent conversations about kube-apiserver and enabled featureset version,
 it is possible to incorporate other version relating to the cluster, such as
 minimum kubelet version, maximum kubelet version, and enabled featureset version.
 
+#### Credentials
+
+An array of credentials to access the cluster. Each credential should be assigned 
+to a different consumer. Each credential has: 
+- a **consumer** field to indicate which consumer it is assigned to.
```

Comment:
1/ The "consumer" in this context specifically refers to credential consumers, entities that require access credentials. While the ClusterProfile API may have multiple consumers, the "consumer" field within the credentials section is intended to specify which credential is assigned to which credential consumer. Maybe we can rename this field to provide more clarity? ie, credentialConsumer
2,3/ There isn't a predefined or global list of credential consumers because they can vary per ClusterProfile. Each cluster may have different entities that need access, so the list of credential consumers is defined within each ClusterProfile as needed.

WDYT?

Reply:
I don't think credentialConsumer is a concept we should have. Either we have a Consumer and we define it or we revisit the whole thing to avoid the notion of a "consumer".

Otherwise we will have weird definitions for one-off and a very disconnected approach.

2,3/ fair; I'm trying to define Consumer, but keeping some openness to it in my branch. It's kinda akin an interface. "Consumer are xxx, and for this design we need at least X,Y,Z; rest may be necessary for other KEPs"
https://github.com/ryanzhang-oss/enhancements/compare/KEP4322-credential...corentone:enhancements:KEP4322-credential